libraries/RC_Channel/RC_Channel.cpp

```diff
@@ -418,12 +418,28 @@ bool RC_Channel::read_6pos_switch(int8_t& position)
         return false;  // This is an error condition
     }
 
+#if FLIGHT_MODE_COUNT <= 6
```

Comment:
This wont work for other instances where the 6 position switch is read. ie. `bool RC_Channel::read_aux()`